src/WindowTitleBar/CloneButton.tsx

```diff
@@ -0,0 +1,173 @@
+import React from "react";
+import {FinsembleWindow, SpawnParams, WindowBounds} from "@finsemble/finsemble-core";
+
+type CloneButtonProps = {
+
+}
+
+export const CloneButton: React.FunctionComponent<CloneButtonProps> = () => {
+
+	/**
+	 * Clones an app
+	 * @param bounds
+	 * @param addToTabGroup
+	 * @param staggerPosition
+	 */
+	const cloneApp = async (bounds?:WindowBounds, addToTabGroup?:boolean, staggerPosition:number = 25) => {
+		let {component, spawnParams,componentState} = await getComponentData(finsembleWindow);
+		if(bounds) {
+			delete spawnParams.top;
+			delete spawnParams.left;
+			delete spawnParams.width;
+			delete spawnParams.height;
+			spawnParams = {...spawnParams, ...bounds}
+		}
+
+		if(!bounds && staggerPosition) {
+			// Can this always be assumed to be a number?
+			spawnParams.top = spawnParams.top ? spawnParams.top as number + staggerPosition : staggerPosition;
+			spawnParams.left = spawnParams.left ? spawnParams.left as number + staggerPosition : staggerPosition;
+		}
+
+		const {response} = await FSBL.Clients.AppsClient.spawn(component, {...spawnParams, addToWorkspace: true});
+		// Race condition
+		prepareComponentState(spawnParams.name as string,componentState)
+
+		if(response?.finWindow) {
+			// setComponentState(response.finWindow, componentState);
+			if(addToTabGroup) {
+				addWindowToTabGroup(response.finWindow.name)
+			}
+		}
+	}
+
+	/**
+	 * Finds local window's tab collection and add's clone to it if it exists. Creates a new tab collection otherwise.
+	 * @param name
+	 */
+	const addWindowToTabGroup = async (name) => {
+		const { err, data } = await FSBL.Clients.WindowClient.getTabCollectionForWindow({windowName: finsembleWindow.name});
+		if(err) {
+			await FSBL.Clients.WindowClient.createTabCollection({windowNames: [finsembleWindow.name, name], promotedTab:name});
+			return;
+		}
+
+		if(data) {
+			await FSBL.Clients.WindowClient.addToTabCollection({
+				tabCollectionId: data.id,
+				windowNames: [name],
+				promotedTab: name
+			});
+		}
+	}
+
+	/**
+	 * Too slow. Linker does not change.
+	 */
+	const setComponentState = (fsblWindow: FinsembleWindow, componentState: Object) => {
+		const fields:any[] = [];
+		Object.keys(componentState).forEach((key) => {
+			fields.push({field:key, value:componentState[key]})
+		})
+
+		fsblWindow.setComponentState({fields})
+	}
+
+	/**
+	 * Originally used to set component state before spawn but set it immediately after spawn - potential race condition.
+	 * @param windowName
+	 * @param componentState
+	 */
+	const prepareComponentState = async (windowName:string, componentState:any) => {
```

Comment:
Pure musing, but I wonder if they will ever need to mutate the state for the clone... That could be possible for an API call (same way as we do with workspaces, pass in a fn that does the mutation), but not for a built-in copy button. Doing anything about this would be premature without client feedback however.